cli/flox/src/commands/init.rs

```diff
@@ -0,0 +1,353 @@
+
+        for hook in hooks {
+            // Run hooks if we can't prompt
+            if hook.should_run(dir)
+                && (self.auto_setup || !Dialog::can_prompt() || hook.prompt_user()?)
+            {
+                customizations.push(hook.get_init_customization())
+            }
+        }
+
+        Ok(Self::combine_customizations(customizations))
+    }
+
+    /// Deduplicate packages and concatenate hooks into a single string
+    fn combine_customizations(customizations: Vec<InitCustomization>) -> InitCustomization {
+        let mut hooks: Vec<String> = vec![];
+        // Deduplicate packages with a set
+        let mut packages_set = HashSet::<PackageToInstall>::new();
+        for customization in customizations {
+            if let Some(packages) = customization.packages {
+                packages_set.extend(packages)
+            }
+            if let Some(hook) = customization.hook {
+                hooks.push(hook)
+            }
+        }
+
+        let hook = (!hooks.is_empty()).then(|| {
+            formatdoc! {"
+                # Autogenerated by flox
+
+                {}
+
+                # End autogenerated by flox", hooks.join("\n\n")}
+        });
+
+        let packages = (!packages_set.is_empty())
+            .then(|| packages_set.into_iter().collect::<Vec<PackageToInstall>>());
+
+        InitCustomization { hook, packages }
+    }
+}
+
+trait InitHook {
+    fn should_run(&self, path: &Path) -> bool;
+
+    fn prompt_user(&self) -> Result<bool>;
+
+    fn get_init_customization(&self) -> InitCustomization;
+}
+
+struct Requirements;
+
+impl InitHook for Requirements {
+    fn should_run(&self, path: &Path) -> bool {
+        path.join("requirements.txt").exists()
+    }
+
+    fn prompt_user(&self) -> Result<bool> {
+        let message = formatdoc! {"
+            Flox detected a requirements.txt
+
+            Python projects typically need:
+            * python, pip
+            * A Python virtual environment to install dependencies into
+
+            Would you like Flox to apply the standard Python environment?
+            You can always revisit the environment's declaration with 'flox edit'
+        "};
+
+        let help = "Use '--auto-setup' to apply Flox recommendations in the future.";
+
+        let dialog = Dialog {
+            message: &message,
+            help_message: Some(help),
+            typed: Select {
+                options: ["Yes (python 3.11)", "No", "Show suggested modifications"].to_vec(),
+            },
+        };
+
+        let (mut choice, _) = dialog.raw_prompt()?;
+
+        if choice == 2 {
+            let message = formatdoc! {"
+
+            {}
+            Would you like Flox to apply these modifications?
+            You can always revisit the environment's declaration with 'flox edit'
+        ", format_customization(&self.get_init_customization())?};
+
+            let help = "Use '--auto-setup' to apply Flox recommendations in the future.";
```

Comment:
Non-blocking - Perhaps break this out and generate it once after all init customizations have been applied... IFF any have been applied.  Somewhere around init.rs:95?

Reply:
Grepping for `help_message: Some(` we have a few similar help messages - think we should be consistent here, or make it conditional on the user selecting `Yes`?

Reply:
Understood. My rationale was to limit duplication and discrepancies once we have more of these.  If there's 3 hooks applied, and they all show this message, and there's typos between etc... Trying to avoid that kind of thing.  
Once there are more, and the user selects `Yes` for some, and `No` for others, it gets a lot messier in that `--auto-setup` is an all or nothing proposition.  Not sure how exactly to handle that without inventing a whole scheme to identify each customization.  That certainly sounds like a *tomorrow* problem to me.

Punting on that and a conditional on selecting `Yes` makes sense IMO.  Perhaps a TODO to revisit this scenario when we add the next?

Reply:
Added a TODO